### ISSUE TITLE
Cannot assign a compatible tuple to `json`

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#22244]
**Description:**
$title.

**Steps to reproduce:**
```ballerina
public function main() {
    [string, int] x = ["hello", 1];
    json j = x; // error: incompatible types: expected 'json', found '[string,int]'

    (string|int)[] y = x;
    json k = y; // Works.
}
```

Need to also check the `.toJsonString()` representation for a tuple.
```ballerina
import ballerina/io;

public function main() {
    [string, int] x = ["hello", 1];
    (string|int)[] y = x;

    io:println(y.toJsonString()); // prints `hello 1`, expected `["hello", 1]`
}
```

**Affected Versions:**
1.2.0

**Comment by MaryamZi** [comment]
The `.toJsonString()` representation works as expected on slp5.

**Comment by dulvinw** [comment]
@MaryamZi Do we need to fix this in 1.2.x?

**Comment by MaryamZi** [comment]
> @MaryamZi Do we need to fix this in 1.2.x?

IMO, it should be enough if we fix this on the master branch unless there's a specific requirement.

Assigning a compatible tuple to json doesn't work yet - slalpha1.
